flink-datastream-api/src/main/java/org/apache/flink/datastream/api/context/NonPartitionedRuntimeContext.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.datastream.api.context;
+
+import org.apache.flink.annotation.Experimental;
+
+/**
+ * A {@link NonPartitionedRuntimeContext} contains all execution information unrelated to partition.
+ */
+@Experimental
+public interface NonPartitionedRuntimeContext {}
```

Comment:
The contexts become confusing.

There are currently 5 contexts, and there relationship and differences are unclear.
- RuntimeContext
- PartitionedRuntimeContext
- NonPartitionedRuntimeContext
- NonPartitionedContext
- TwoOutputNonPartitionedContext

I think the information / functionalities provided by these contexts can be categorized into 3 kinds
- Only needed in partitioned context (PartitionedRuntimeContext)
- Only needed in non-partitioned context (NonPartitionedContext & TwoOutputNonPartitionedContext)
- Needed by both partitioned and non partitioned (NonPartitionedRuntimeContext)

Therefore, I'd suggest:
- RuntimeContext for things needed by both partitioned and non-partitioned, which is exactly the current NonPartitionedRuntimeContext
- OneOutputNonPartitionedContext and TwoOutputNonPartitionedContext, both extends RuntimeContext, for things only needed in non-partitioned context, which are the current NonPartitionedContext & TwoOutputNonPartitionedContext
- PartitionedContext, extends RuntimeContext, for things only needed in partitioned context, which is the current RuntimeContext
- The current PartitionedRuntimeContext is no longer needed, as is is neither directly implemented by any concrete classes, nor extended by multiple sub-interfaces/classes.